pkg/apierrors/apierrors.go

```diff
@@ -54,7 +75,21 @@ func IsNotFoundError(err error) bool {
 	var apiErr *APIError
 	if errors.As(err, &apiErr) {
 		switch apiErr.Code {
-		case userNotFoundErrCode:
+		case userNotFoundErrCode, projectNotFoundErrCode:
```

Comment:
Just a suggestion here, but as you can see, the `IsError` functions have all the same definition and the only thing they differ in are the error codes comparisons. Meaning you can simplify this flow by implementing one `IsErrorWithCode` function of the signiture `IsErrorWithCode(err error, code Code)`.